src/sentry/dynamic_sampling/tasks/collect_orgs.py

```diff
@@ -0,0 +1,42 @@
+from sentry_sdk import capture_message, set_extra
+
+from sentry import options
+from sentry.dynamic_sampling.tasks.common import GetActiveOrgs, TimedIterator, TimeoutException
+from sentry.dynamic_sampling.tasks.constants import MAX_PROJECTS_PER_QUERY, MAX_SECONDS
+from sentry.dynamic_sampling.tasks.logging import log_task_execution, log_task_timeout
+from sentry.dynamic_sampling.tasks.task_context import TaskContext
+from sentry.dynamic_sampling.tasks.utils import dynamic_sampling_task
+from sentry.tasks.base import instrumented_task
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.collect_orgs",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=2 * 60 * 60,
+    time_limit=2 * 60 * 60 + 5,
+)
+@dynamic_sampling_task
+def collect_orgs() -> None:
+
+    enabled = options.get("dynamic_sampling.tasks.collect_orgs") or False
+
+    if not enabled:
+        return
+
+    context = TaskContext("sentry.dynamic_sampling.tasks.collect_orgs", MAX_SECONDS)
+    iterator_name = GetActiveOrgs.__name__
+    try:
+        for orgs in TimedIterator(
+            context, iterator_name, GetActiveOrgs(max_projects=MAX_PROJECTS_PER_QUERY)
+        ):
+            pass
```

Comment:
For the next PR lets add some business logic here that fetches the data, so that we can derive a better upper bound.

Reply:
The point of this task is to measure GetOrgs so I wouldn't add anything on top.
The data is fetched and returned in orgs but that's all.